enzyme/Enzyme/PreserveNVVM.cpp

```diff
@@ -368,6 +368,106 @@ bool preserveNVVM(bool Begin, Function &F) {
       "__enzyme_register_derivative";
   constexpr static const char splitderivative_handler_name[] =
       "__enzyme_register_splitderivative";
+
+  if (Begin)
+    if (GlobalVariable *GA =
+            F.getParent()->getGlobalVariable("llvm.global.annotations")) {
+      if (GA->hasInitializer()) {
+        auto AOp = GA->getInitializer();
+        // all metadata are stored in an array of struct of metadata
+        if (ConstantArray *CA = dyn_cast<ConstantArray>(AOp)) {
+          // so iterate over the operands
+          SmallVector<Constant *, 1> replacements;
+          for (Value *CAOp : CA->operands()) {
+            // get the struct, which holds a pointer to the annotated function
+            // as first field, and the annotation as second field
+            if (ConstantStruct *CS = dyn_cast<ConstantStruct>(CAOp)) {
```

Comment:
Can we reduce the indentation level here with something like this
```
ConstantStruct *CS = dyn_cast<ConstantStruct>(CAOp
if (!CS)
   continue;
```

same for the following lines.